### PR TITLE
Source Chargebee: Migrate to YAML

### PR DESCRIPTION
**Pull request by artem1205**
## What
Resolving https://github.com/airbytehq/airbyte/issues/21389

## How
*Describe the solution*

## Recommended reading order
2. `y.python`

## 🚨 User Impact 🚨
schemas field type changed; refresh needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

### CODE REVIEW

**Review by lazebnyi — APPROVED** [review]

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/chargebee.yaml
Comment (girarda) [inline airbyte-integrations/connectors/source-chargebee/source_chargebee/chargebee.yaml:86]: this won't do anything right?

**Review by girarda — CHANGES_REQUESTED** [review]
mostly just a few questions. Also want to make sure we're not blocked on being able to disable streams in the discover operation as [per our discussion](https://airbytehq-team.slack.com/archives/C02UF50V9HA/p1674225973082389)

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/chargebee.yaml
Comment (artem1205) [reply airbyte-integrations/connectors/source-chargebee/source_chargebee/chargebee.yaml:42]: Chargebee API works weird, because it have an option to set both `updated_after` and `updated_before` parameters in request, but in result it returns all the records (logical `OR` instead of `AND`).
So I need to use only `updated_after` in request and then pagination.

Comment (artem1205) [reply airbyte-integrations/connectors/source-chargebee/source_chargebee/chargebee.yaml:42]: Chargebee API works weird, as if we pass both `updated_after` and `updated_before` parameters in request, it returns all records (logical `OR` instead of `AND` is used). Thay have special filter `between`, which should looks like `updated[between]=[start_datetime, end_datetime]`, but we have no option for this case in low-code.
Right now it is enough to use just updated_after and pagination

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/chargebee.yaml
Comment (artem1205) [reply airbyte-integrations/connectors/source-chargebee/source_chargebee/chargebee.yaml:86]: deleted, extra line

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/chargebee.yaml
Comment (artem1205) [reply airbyte-integrations/connectors/source-chargebee/source_chargebee/chargebee.yaml:8]: updated

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/chargebee.yaml
Comment (artem1205) [reply airbyte-integrations/connectors/source-chargebee/source_chargebee/chargebee.yaml:38]: I need datetime_format in second, if I use just `now_utc()` [error is raised here](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/declarative/datetime/datetime_parser.py#L26-L27)

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/chargebee.yaml
Comment (artem1205) [reply airbyte-integrations/connectors/source-chargebee/source_chargebee/chargebee.yaml:35]: I need datetime_format in second, if I use just `now_utc()` [error is raised here](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/declarative/datetime/datetime_parser.py#L26-L27)

**Comment by artem1205** [comment]
@girarda , I've `disabled` streams by using special [error handler](https://github.com/airbytehq/airbyte/blob/8e7fc0e965432691212a752d6e0478f0cc7a8377/airbyte-integrations/connectors/source-chargebee/source_chargebee/chargebee.yaml#L25-L27)

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/chargebee.yaml
Comment (girarda) [reply airbyte-integrations/connectors/source-chargebee/source_chargebee/chargebee.yaml:40]: @artem1205 you should be able to do this by removing the `start_time_option` and adding a request parameter to the requester
```
request_parameters:
 "updated[between]": "\"[{{stream_slice['start_time']}}, {{stream_slice['end_time']}}]\""
```

to produce a request like
` "url": "https://airbyte-test.chargebee.com/api/v2/subscriptions?sort_by%5Basc%5D=updated_at&include_deleted=true&updated%5Bbetween%5D=%5B1621695464%2C+1675133396%5D"`

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/chargebee.yaml
Comment (girarda) [reply airbyte-integrations/connectors/source-chargebee/source_chargebee/chargebee.yaml:55]: thanks for adding a comment

**Review by girarda — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/chargebee.yaml
Comment (artem1205) [reply airbyte-integrations/connectors/source-chargebee/source_chargebee/chargebee.yaml]: updated

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/components.py
Comment (sherifnada) [inline airbyte-integrations/connectors/source-chargebee/source_chargebee/components.py:13]: could you create an issue for this custom component like described [here](https://docs.google.com/document/d/1LB4gBk0slHj3zNmq1Qtj1U7VkKyTN9eFsVldgQ8T4HQ/edit?usp=drivesdk)?

File: airbyte-integrations/connectors/source-chargebee/acceptance-test-config.yml
Comment (sherifnada) [inline airbyte-integrations/connectors/source-chargebee/acceptance-test-config.yml:19]: Can you clarify if this is going to break existing customers in production? If yes then how is that being handled?

**Review by sherifnada — CHANGES_REQUESTED** [review]
requesting changes because of backwards incompatible changes on a GA connector. How will breaking changes impact customers?

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/components.py
Comment (artem1205) [reply airbyte-integrations/connectors/source-chargebee/source_chargebee/components.py:13]: done https://github.com/airbytehq/airbyte/issues/23134

File: airbyte-integrations/connectors/source-chargebee/setup.py
Comment (artem1205) [reply airbyte-integrations/connectors/source-chargebee/setup.py]: update to 0.29

File: airbyte-integrations/connectors/source-chargebee/acceptance-test-config.yml
Comment (artem1205) [reply airbyte-integrations/connectors/source-chargebee/acceptance-test-config.yml]: I'll contact cloud support team to communicate with customers (for schema update)

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/manifest.yaml
Comment (artem1205) [reply airbyte-integrations/connectors/source-chargebee/source_chargebee/manifest.yaml:64]: done

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/manifest.yaml
Comment (artem1205) [reply airbyte-integrations/connectors/source-chargebee/source_chargebee/manifest.yaml:80]: updated

**Review by sherifnada — APPROVED** [review]

File: airbyte-integrations/connectors/source-chargebee/setup.py
Comment (girarda) [reply airbyte-integrations/connectors/source-chargebee/setup.py:9]: @artem1205 I don't see where the version is pinned

**Review by girarda — APPROVED** [review]

**Review by girarda — APPROVED** [review]
approved pending pinning the CDK version to a minor release

**Pull request merged**
